tests/tools/oecertdump/host/sgx_quote.cpp

```diff
@@ -111,6 +136,47 @@ void output_certificate_chain(const uint8_t* data, size_t data_len)
         result = ParseSGXExtensions(
             &leaf_cert, buffer, &buffer_size, &extension_info);
     }
+    printf(
+        "\n    parsed qe certificate extension (%s) {\n",
+        SGX_EXTENSION_OID_STR);
+    printf("        ppid (hex): ");
+    oe_hex_dump(extension_info.ppid, OE_COUNTOF(extension_info.ppid));
+    printf("        comp_svn (hex): ");
+    oe_hex_dump(extension_info.comp_svn, OE_COUNTOF(extension_info.comp_svn));
+    printf("        pce_svn: 0x%x\n", extension_info.pce_svn);
+    printf("        cpu_svn (hex): ");
+    oe_hex_dump(extension_info.cpu_svn, OE_COUNTOF(extension_info.cpu_svn));
+    printf("        pce_id (hex): ");
+    oe_hex_dump(extension_info.pce_id, OE_COUNTOF(extension_info.pce_id));
+    printf("        fmspc (hex): ");
+    oe_hex_dump(extension_info.fmspc, OE_COUNTOF(extension_info.fmspc));
+    printf("        sgx_type: %d\n", extension_info.sgx_type);
+    printf(
+        "        opt_dynamic_platform: %s\n",
+        extension_info.opt_dynamic_platform ? "true" : "false");
+    printf(
+        "        opt_cached_keys: %s\n",
+        extension_info.opt_cached_keys ? "true" : "false");
+    printf("    } qe cert extension \n");
+done:
+    free(buffer);
+    oe_cert_chain_free(&cert_chain);
+    oe_cert_free(&leaf_cert);
+}
+
+void output_certificate_chain(
+    const uint8_t* data,
+    size_t data_len,
+    bool is_report_buffer)
+{
+#if defined(__linux__)
```

Comment:
Why is this only for Linux?

Reply:
This oecertdump tool is for Linux only.

Reply:
Why?

Reply:
Sorry for being rash. 
I've just tried to implemented that in windows build. But it seems the openssl print functions like `x509_print_ex_fp()`, which outputs the decoded PEM, require OPENSSL_Applink. It can be included at <openssl/applink.c>. But then complier will then throw error (warning treated as error) that the `fopen()` in `applink.c` is not safe.
Do you think we can implemented this feature for Linux in this PR first? In windows, the oecertdump now will output the completed PEM string that can be decoded manually.

Reply:
I think the functionality should be the same between Linux and Windows.

Reply:
<openssl/applink.c> uses unsafe APIs which is against our compiling flag settings. I'd recommend removing "treat warning as error" flag in Windows build and carefully verifying we are using the unsafe API correctly. Otherwise, since this tool is used internally for our infrastructure daily testing (at time moment Linux VMs), I'd be OK to spin off another task to address the minor inefficiency in Windows and merge this PR as is.